test/Microsoft.AspNetCore.Hosting.FunctionalTests/ShutdownTests.cs

```diff
@@ -43,6 +44,8 @@ public ShutdownTests(ITestOutputHelper output) : base(output)
                     PublishApplicationBeforeDeployment = true
                 };
 
+                deploymentParameters.EnvironmentVariables.Add(new KeyValuePair<string, string>("ASPNETCORE_STARTMECHANIC", "Run"));
```

Comment:
Should I change it to an `IDictionary` as part of this change as well? I'll need to check if this breaks things downstream though.